### DISCUSSION TITLE
When using langchain+miluvs, I encountered a "KeyError" when inserting a document. Can you tell me what the reason might be? Thank you.

### DISCUSSION THREAD

**Discussion by zhangnian** [milvus-io/milvus#30569] [Q&A and General discussion]
```
langchain            0.1.5
langchain-community  0.0.17
langchain-core       0.1.18
pymilvus             2.3.6
PyMuPDF              1.23.21
```

```python
from langchain_community.document_loaders import PyMuPDFLoader
from langchain.text_splitter import RecursiveCharacterTextSplitter
from langchain_community.vectorstores import Milvus
from langchain_community.embeddings import BaichuanTextEmbeddings

file_path = ''"招标代理政府采购相关法律法规 (19年定稿)(1).pdf"
loader = PyMuPDFLoader(file_path)
documents= loader.load()

text_splitter = RecursiveCharacterTextSplitter(chunk_size=500, chunk_overlap=50)
docs = text_splitter.split_documents(documents)

embeddings = BaichuanTextEmbeddings(baichuan_api_key='sk-a7bc6fcd9c4468bacc595601388e6a22')
res = Milvus.from_documents(docs[:3], embeddings, connection_args={"host": 'vm', "port": "19530"})
```

Exception happens:

KeyError: 'pk'

**Comment by yhmo** [comment]
Recently some commits were submitted, the latest commit added a new parameter "auto_id" for the MIlvusVectorstore. The default value is "False". Since auto_id is False, it requires the client to provide a primary key value every time call insert(). (the primary key field default name is "pk"). However, the "docs" generated by text_splitter.split_documents() doesn't have a metadata named "pk", so it reports this error.


A PR is submitted to fix this defect: https://github.com/langchain-ai/langchain/pull/17021

Before the fix PR gets merged, you can workaround by downgrading the langchain version to <=v0.1.4
